src/actions/NotificationActions.js

```diff
@@ -38,3 +48,186 @@ export const enableNotifications = (currencyCode: string, hours: string, enabled
+          createEvents.push(newPriceChangeEvent(activeCurrencyInfos[i], defaultIsoFiat, setting[1] === true, setting[24] === true))
+        }
+      }
+    }
+  }
+
+  if (createEvents.length > 0) {
+    v2Settings = await setDeviceSettings(deviceId, { createEvents })
+  }
+
+  dispatch({
+    type: 'PRICE_CHANGE_NOTIFICATIONS_UPDATE',
+    data: serverSettingsToState(v2Settings)
+  })
+}
+
+export const serverSettingsToState = (settings: $Call<typeof asDevicePayload>): PriceChangeNotificationSettings => {
+  const data: PriceChangeNotificationSettings = { ignorePriceChanges: settings.ignorePriceChanges }
+
+  for (const event: PushEventStatus of settings.events) {
+    if (event.state !== 'waiting') continue
+    const trigger = asMaybe(asPriceChangeTrigger)(event.trigger)
+    if (trigger == null) continue
+
+    data[trigger.pluginId] = {
+      eventId: event.eventId,
+      currencyPair: trigger.currencyPair,
+      dailyChange: trigger.dailyChange,
+      hourlyChange: trigger.hourlyChange
+    }
+  }
+
+  return data
+}
+
+export const getDeviceSettings = async (deviceId: string): Promise<$Call<typeof asDevicePayload>> => {
+  const body = {
+    apiKey: ENV.AIRBITZ_API_KEY,
+    deviceId
+  }
+  const opts = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  }
+  const response = await fetchPush('v2/device/', opts)
+
+  return asDevicePayload(await response.json())
+}
+
+export const setDeviceSettings = async (deviceId: string, data: DeviceUpdatePayload): Promise<$Call<typeof asDevicePayload>> => {
+  const deviceToken = await messaging()
+    .getToken()
+    .catch(e => console.log('Failed to fetch firebase device token.', e.message))
+
+  const body = {
+    apiKey: ENV.AIRBITZ_API_KEY,
+    deviceId,
+    deviceToken,
+    data
+  }
+  const opts = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  }
+
+  const response = await fetchPush('v2/device/update/', opts)
+
+  return asDevicePayload(await response.json())
+}
+
+export const newPriceChangeEvent = (
+  currencyInfo: EdgeCurrencyInfo,
+  isoFiatCurrencyCode: string,
+  hourlyChangeEnabled: boolean,
+  dailyChangeEnabled: boolean
+): NewPushEvent => {
+  const { currencyCode, displayName } = currencyInfo
+
+  const fiatDenomination = getDenomFromIsoCode(isoFiatCurrencyCode.replace('iso:', ''))
+  const fiatSymbol = fiatDenomination.symbol ?? ''
+
+  const pushMessage = {
+    title: s.strings.price_alert,
+    body: `${sprintf(s.strings.notification_price_change, displayName, currencyCode, fiatSymbol)}`,
+    data: {
+      localizedPriceChangeDailyMessage: `${sprintf(s.strings.notification_daily_price_change, displayName, currencyCode, fiatSymbol)}`,
+      localizedPriceChangeHourlyMessage: `${sprintf(s.strings.notification_hourly_price_change, displayName, currencyCode, fiatSymbol)}`
```

Comment:
I understand this is meant for the app to user, however, because each price change event is actually two events with different localized strings (one is singular the other plural). I added a commit to the push server to parse this special case. 

I'm open to another suggestion to get two different strings in. The programatic way on the push server of recreating strings could cause some weirdness with word order and plurals for translations which is why I bothered to code around it. Another idea could be to create one long string and split out correct one using aggressive hashtags:

body: "###HOURLY###bitcoin ha subido un 3% en la última hora###HOURLY######DAILY###Bitcoin ha subido un 10% en las últimas 24 horas###DAILY###